apluslms_roman/backends/__init__.py

```diff
@@ -67,23 +67,31 @@ def __str__(self):
 
 
 class BuildResult:
-    __slots__ = ('code', 'error', 'step')
+    __slots__ = ('step', 'ok', 'code', 'error')
 
-    def __init__(self, code=0, error=None, step=None):
-        self.code = code
-        self.error = error
-        self.step = step
-        assert self.ok or step is not None, "step is required for failed result"
+    def __init__(self, ok=True, *, code=0, error=None, step=None):
+        self.step, self.ok, self.code, self.error = (
```

Comment:
Can these be one per line and can we have a) assert to test that ok is false when code is != 0, or b) that ok is None by default and set to `code != 0` in that case.